.vib/mongodb-sharded/ginkgo/mongodb_test.go

```diff
@@ -0,0 +1,125 @@
+package mongodb_sharded_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	utils "github.com/bitnami/charts/.vib/common-tests/ginkgo-utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	PollingInterval = 1 * time.Second
+)
+
+var _ = Describe("MongoDB Sharded", Ordered, func() {
+	var c *kubernetes.Clientset
+	var ctx context.Context
+	var cancel context.CancelFunc
+
+	BeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.Background())
+
+		conf := utils.MustBuildClusterConfig(kubeconfig)
+		c = kubernetes.NewForConfigOrDie(conf)
+	})
+
+	When("a database is created and Mongodb Shards are scaled down to 0 replicas and back up", func() {
+		It("should have access to the created database", func() {
+
+			getAvailableReplicas := func(ss *appsv1.StatefulSet) int32 { return ss.Status.AvailableReplicas }
+			getSucceededJobs := func(j *batchv1.Job) int32 { return j.Status.Succeeded }
+
+			for i := 0; i < shards; i++ {
+				By(fmt.Sprintf("checking all the shard %d replicas are available", i))
+				shardName := fmt.Sprintf("%s-shard%d-data", releaseName, i)
+				ss, err := c.AppsV1().StatefulSets(namespace).Get(ctx, shardName, metav1.GetOptions{})
```

Comment:
You're using `metav1.GetOptions{}` on every `xxx.Get` function. Please create a variable and reuse it on every call:
```golang
			getOpts := metav1.GetOptions{}
			(...)
			foo, errr := c.AppsV1().XXX.Get(ctx, xxx, getOpts)
```